### PR TITLE
fixed toString()'s and added Address skeleton matching()'s

### PR DESCRIPTION
**Pull request by vardahoth**

### CODE REVIEW

File: src/com/cft/contactmerge/contact/Address.java [hunk 1]
```diff
@@ -77,6 +79,40 @@ public void setZip(String zip) {
      *************************************************** Sub-Methods ***************************************************
      *******************************************************************************************************************/
     public String toString() {
-        return getStreetAddress() + ", " + getApartment() + ", " + getCity() + ", " +
-                getState() + ", " + getCountry() + ", " + getZip(); }
+        StringBuilder fullAddress = new StringBuilder();
+        String[] addressParts = {getStreetAddress(), getApartment(), getCity(), getState(), getCountry(), getZip()};
```
Comment (mbilodeau6) [inline src/com/cft/contactmerge/contact/Address.java]: I just had to write something like this myself and found a nice version on StackOverflow… It avoids having the go back and remove the extra delimiter.     

// from https://stackoverflow.com/questions/4930939/preferred-idiom-for-joining-a-collection-of-strings-in-java
    private String joinCollectionOfStrings(Collection<String> collection, String delimeter)
    {
        Iterator<String> iterator = collection.iterator();
        StringBuilder sb = new StringBuilder();

        if (iterator.hasNext()) {
            sb.append(iterator.next());

            while (iterator.hasNext()) {
                sb.append(delimeter).append(iterator.next());
            }
        }

        return sb.toString();
    }

File: src/com/cft/contactmerge/contact/Address.java [hunk 2]
```diff
@@ -77,6 +79,40 @@ public void setZip(String zip) {
...
+        }
+
+        if (!fullAddress.toString().isEmpty()){
+            fullAddress.delete(fullAddress.length() - 2, fullAddress.length());
+        }
+
+        return fullAddress.toString();
+    }
+
+    public AnswerType isAddressMatch(Address otherAddress){
```
Comment (mbilodeau6) [inline src/com/cft/contactmerge/contact/Address.java]: My current thoughts:
1. All address parts except apartment and zip are required. We won't add an address without these and so you shouldn't have to worry about getting and address without StreetAddress, City, State, or Country.
2. It is only a yes if all parts match. 
3. There are only three maybe cases I can think of that will result in a Address match of maybe : a) the StreetAddress match is a maybe and everything else is a yes; b) the StreetAddress is a yes or maybe and the Apartment is missing on one of the addresses being compared... everything else matches; c) the StreetAddress is a yes or maybe and the Zip is a maybe or missing... everything else matches.

File: src/com/cft/contactmerge/contact/Name.java [hunk 3]
```diff
@@ -78,6 +78,14 @@ public AnswerType isMatch(Name otherProperty) {
     }
 
     public String toString() {
-        return getFirstName() + ", " + getMiddleName() + ", " + getLastName() + ", " + getPrefix() + ", " + getSuffix();
+        if (!getFirstName().isEmpty() && !getLastName().isEmpty()){
```
[the remark this answers is not in the document]
Comment (vardahoth) [reply src/com/cft/contactmerge/contact/Name.java]: Will add this then as a pre-condition.

File: src/com/cft/contactmerge/contact/Address.java (hunk 2 above)
Comment (vardahoth) [reply src/com/cft/contactmerge/contact/Address.java]: Thanks.

File: src/com/cft/contactmerge/contact/Address.java (hunk 1 above)
Comment (vardahoth) [reply src/com/cft/contactmerge/contact/Address.java]: Nice, I'll use this in the next pull.

**Pull request merged**